tda/streaming.py

```diff
@@ -60,33 +60,83 @@ def __init__(self, raw_msg, json_parse_exception, *args, **kwargs):
         self.json_parse_exception = json_parse_exception
 
 
-class _Handler:
-    def __init__(self, func, field_enum_type):
-        self._func = func
-        self._field_enum_type = field_enum_type
-
-    def __call__(self, *args, **kwargs):
-        return self._func(*args, **kwargs)
-
-    def label_message(self, msg):
+def _handler(field_enum_type):
+    '''Returns a function which returns a relabeled copy of the argument according to field_enum_type'''
+    def __handler(msg):
         if 'content' in msg:
             new_msg = copy.deepcopy(msg)
             for idx in range(len(msg['content'])):
-                self._field_enum_type.relabel_message(msg['content'][idx],
-                                                      new_msg['content'][idx])
+                field_enum_type.relabel_message(msg['content'][idx],
+                                                new_msg['content'][idx])
             return new_msg
         else:
             return msg
+    return __handler
+
+
+def _bookhandler(field_enum_type):
+    _top_level_handler = _handler(field_enum_type)
+
+    def __handler(msg):
+        # Relabel top-level fields
+        new_msg = _top_level_handler(msg)
+
+        # Relabel bids
+        for content in new_msg['content']:
+            if 'BIDS' in content:
+                for bid in content['BIDS']:
+                    # Relabel top-level bids
+                    StreamClient.BidFields.relabel_message(bid, bid)
+
+                    # Relabel per-exchange bids
+                    for e_bid in bid['BIDS']:
+                        StreamClient.PerExchangeBidFields.relabel_message(
+                            e_bid, e_bid)
+
+        # Relabel asks
+        for content in new_msg['content']:
+            if 'ASKS' in content:
+                for ask in content['ASKS']:
+                    # Relabel top-level asks
+                    StreamClient.AskFields.relabel_message(ask, ask)
+
+                    # Relabel per-exchange bids
+                    for e_ask in ask['ASKS']:
+                        StreamClient.PerExchangeAskFields.relabel_message(
+                            e_ask, e_ask)
+
+        return new_msg
+
+    return __handler
 
 
 class StreamClient(EnumEnforcer):
 
-    def __init__(self, client, *, account_id=None, enforce_enums=True, ssl_context=None):
+    def __init__(self, client, *, account_id=None,
+                 enforce_enums=True, ssl_context=None):
         super().__init__(enforce_enums)
 
         self._ssl_context = ssl_context
         self._client = client
 
+        self._service_to_handler = {
```

Comment:
I'm finding it a little confusing keeping track of what `_service_to_handler` is doing and what `_handlers` is doing. Any way we can fuse the two? Maybe make the above methods into classes with a list of callbacks that get called on handling?